resorts/views/resort_list.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+# General imports
+from collections import defaultdict
+from django.core.exceptions import ObjectDoesNotExist
+
+# Import Resort and ConditionsReport models
+from ..models import Resort, Conditions
+
+# Imports for using Google Maps
+import googlemaps
+import json
+
+# Google Maps requires a client key to work
+GMAPS = googlemaps.Client(key='AIzaSyBRrCgnGFkdRY-Z1hX6xaxoUFBczNI2664')
+
+# Constants
+NUMBER_TO_DISPLAY = 5
+DEFAULT_ORDER_ON = 'snow_in_past_24h'
+
+def get_resort_list(selected_resorts, user_address = "", number_to_display = NUMBER_TO_DISPLAY, order_on = DEFAULT_ORDER_ON):
+    """ Given a list of selected resorts, returns an ordered list of resorts and their conditions for display.
+        If a user_address is supplied, also returns driving time and distance information.
+    
+    Args:
+        selected_resorts (list): Queryset of Resorts
+        user_address (str, optional): Indicates the user address
+        number_to_display (int, optional): How many resorts we want to display
+        order_on (str, optional): How we want to order the resort list
+
+    Returns:
+        list: List of Resort dictionaries with name, id, condition information, and optional driving information
+
+    """
+    resorts_list = [x for x in selected_resorts.values('name', 'id')]
+
+    if user_address != "":
+        use_googlemaps(user_address, selected_resorts, resorts_list)
+
+    get_conditions(resorts_list) 
+    return(order_resorts(resorts_list, number_to_display, order_on))
+
+
+def use_googlemaps(address, selected_resorts, resorts_list):
+    """ We use the Google Maps API to get driving distances and times for the user.
+        Google Maps returns a JSON response that we must parse.
+        For how to response object looks, see: https://developers.google.com/maps/documentation/distance-matrix/intro.
+        We grab the distance, chop off the trailing units using [:-3], and get rid of commas.
```

Comment:
These comments may be better placed next to the specific line. Docstring should be used to describe the overall functionality and effects